src/order/countable_dense_linear_order.lean

```diff
@@ -182,38 +174,40 @@ open partial_iso
 
 variables (α β)
 
-/-- Any countable linear order embeds in any nonempty dense linear order without endpoints. -/
-def embedding_from_countable_to_dense
-  [encodable α] [densely_ordered β] [no_min_order β] [no_max_order β] [nonempty β] :
-  α ↪o β :=
-let our_ideal : ideal (partial_iso α β) := ideal_of_cofinals default $ defined_at_left β in
-let F := λ a, fun_of_ideal a our_ideal (cofinal_meets_ideal_of_cofinals _ _ a) in
-order_embedding.of_strict_mono (λ a, (F a).val)
+/-- Any countable order embeds in any nontrivial dense linear order. -/
+theorem embedding_from_countable_to_dense [encodable α] [densely_ordered β] [nontrivial β] :
+  nonempty (α ↪o β) :=
 begin
-  intros a₁ a₂,
-  rcases (F a₁).property with ⟨f, hf, ha₁⟩,
-  rcases (F a₂).property with ⟨g, hg, ha₂⟩,
+  rcases exists_pair_lt β with ⟨x, y, hxy⟩,
+  cases exists_between hxy with a ha,
+  haveI : nonempty (set.Ioo x y) := ⟨⟨a, ha⟩⟩,
+  let our_ideal : ideal (partial_iso α _) :=
+    ideal_of_cofinals default (defined_at_left (set.Ioo x y)),
+  let F := λ a, fun_of_ideal a our_ideal (cofinal_meets_ideal_of_cofinals _ _ a),
+  refine ⟨rel_embedding.trans (order_embedding.of_strict_mono (λ a, (F a).val) (λ a₁ a₂, _))
+    (order_embedding.subtype _)⟩,
+  rcases (F a₁).prop with ⟨f, hf, ha₁⟩,
+  rcases (F a₂).prop with ⟨g, hg, ha₂⟩,
   rcases our_ideal.directed _ hf _ hg with ⟨m, hm, fm, gm⟩,
-  exact (lt_iff_lt_of_cmp_eq_cmp $ m.property (a₁, _) (fm ha₁) (a₂, _) (gm ha₂)).mp
+  exact (lt_iff_lt_of_cmp_eq_cmp $ m.prop (a₁, _) (fm ha₁) (a₂, _) (gm ha₂)).mp
 end
 
 /-- Any two countable dense, nonempty linear orders without endpoints are order isomorphic. -/
-def iso_of_countable_dense
+theorem iso_of_countable_dense
   [encodable α] [densely_ordered α] [no_min_order α] [no_max_order α] [nonempty α]
   [encodable β] [densely_ordered β] [no_min_order β] [no_max_order β] [nonempty β] :
-  α ≃o β :=
+  nonempty (α ≃o β) :=
```

Comment:
Is there a reason for this change?

Reply:
`embedding_from_countable_to_dense` is nonconstructive (the first line of the proof cases on an existential), so it doesn't make much sense for it to be a `def`. The change in `iso_of_countable_dense` was done for consistency. 

I don't believe there's anything of importance that could be recovered from the definitions of either (after all, the theorem is just that an embedding exists, not that it has any extra properties).